frontend/apps/reader/modules/readerlink.lua

```diff
@@ -91,6 +92,138 @@ function ReaderLink:init()
 
     -- delegate gesture listener to readerui, NOP our own
     self.ges_events = nil
+
+    -- Set up buttons for alternative external link handling methods
+    self._external_buttons["01_copy"] = function(this, link_url)
+        return {
+            text = _("Copy"),
+            callback = function()
+                UIManager:close(this.dialog)
+            end,
+        }
+    end
+
+    self._external_buttons["02_qrcode"] = function(this, link_url)
+        return {
+            text = _("Show QR code"),
+            callback = function()
+                UIManager:close(this.dialog)
+                UIManager:show(QRMessage:new{
+                        text = link_url,
+                        width = Device.screen:getWidth(),
+                        height = Device.screen:getHeight()
+                })
+            end
+        }
+    end
+    self._external_buttons["04_browser"] = function(this, link_url)
+        return {
+            text = _("Open in browser"),
+            callback = function()
+                UIManager:close(this.dialog)
+                Device:openLink(link_url)
+            end,
+            show_in_dialog_func = function()
+                if Device:canOpenLink() then
+                    return true
+                end
```

Comment:
I was going to say you could just use `show_in_dialog_func = Device.canOpenLink` as it is itself a straight function.
But I remember @NiLuJe may have said something about using them as methods recently ? (I may have imagining that :/)
Anyway, I don't mind the above, I find it more readable.

Reply:
`Device.canOpenLink` was returning some weird value. It wasn't boolean, from what I could tell. So I had to do this.

Reply:
If there's no argument (and as such no risk of confusion on what the "object" in `self` is, an alias should work?

Reply:
Not sure how to do whatever you're suggesting :)

Reply:
Just `show_in_dialog_func = Device.canOpenLink,` (try it to see if it works).
But the current is just fine.